### ISSUE TITLE
Feature Request: Filter small balances from Statement of Assets

### ISSUE THREAD

**Issue by metal450** [buchen/portfolio#3268]
On the Statement of Assets screen, it would be great if there were an option to filter (not show) any security with a balance below some configurable minimum (i.e. $1, $100, whatever makes sense to you).

In particular, this is useful for crypto, where you often end up with "dust" of lowish values, or needing to hold "gas tokens" for various blockchains. While these holdings should still be tracked along with all others, it would be nice to minimize showing all of them on Statement of Assets, when they have minimal impact on overall net worth.

**Comment by Kansai53** [comment]
agree, a very long list of assets can be hard, I do this by adding all "small things" to a taxonomy, but since values can change, it would be much better if it could just automatically skip securities with very small holdings

**Comment by Kansai53** [comment]
Like how we have a "value not zero" option. Just need i.e. "Value < $100" option, maybe where $100 can be set even just in the xml

**Comment by metal450** [comment]
Where's "value not zero"?

**Comment by chirlu** [comment]
It’s a filter option. The button showing a funnel.

**Comment by metal450** [comment]
It's not there in my filter menu, on Statement of Assets page. The only options are "Entire Portfolio", then an option for each Securities Account, and an option for each Securities Account + corresponding Deposit Account. & at the end I can add my own, but again the only options are accounts. There's no "Value not zero" in my filter menu.

**Comment by metal450** [comment]
Oh I see it in the filter menu on the taxonomies pages, just not on Statement of Assets. Probably because Statement of Assets already doesn't show zero values. Anyway yeah, no way to hide "small values" unfortunately :/

**Comment by metal450** [comment]
Ok, since this has been driving me crazy for awhile now, I tried to see if I could figure out how to implement it myself. With the disclaimer that I've never worked with Eclipse & don't actually know Java, after some stumbling around with the debugger, it looks like I can get the behavior by editing `getElements()` in `StatementOfAssetsViewer.java` & adding:

```
// Filter out items with a value below $100
.filter(e -> e.position == null || e.position.getValuation().getAmount() > 10000)
```

I wish I knew how to add it as a proper filter menu item so it could be included as a real feature, but couldn't figure it out. It seems to have something to do with `PortfolioClientFilter`...but I wasn't able to make much sense of it given how there's virtually no code comments :/

**Comment by chirlu** [comment]
Non-zero entries certainly shouldn’t be silently dropped. People would – rightly – complain that the numbers no longer add up.

If this were to be implemented (I’m generally not sure it is a good idea), it could only be done by *summarizing* small values, i.e. one summary line replacing multiple small entries. It would potentially be necessary to have one summary line per category.

**Comment by metal450** [comment]
> Non-zero entries certainly shouldn’t be silently dropped.

Yeah, I definitely know it's a bad implementation as-is. Was just desperate to get *something* usable for myself, & I don't really have the capability to do it "properly."

> it could only be done by summarizing small values, i.e. one summary line replacing multiple small entries. It would potentially be necessary to have one summary line per category.

One line summarizing small entries would be great. One line summarizing small entries per category would more or less defeat the my use-case tbh, as I'm trying to get rid of the excess lines that don't meaningfully impact the portfolio value, and there's often only one or two per category.
